discord.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/bwmarrin/discordgo"
+	"golang.org/x/exp/slices"
+)
+
+
+func watchOnlineUsers(dg *discordgo.Session, stopWatchingChan <-chan bool) {
+	for {
+		dg.RequestGuildMembers(Guild, "", 100, "members", true)
+		select {
+		case <-stopWatchingChan:
+			break
+		case <-time.After(5 * time.Minute):
+			continue
+		}
+	}
+
+}
+
+func handleCommand(s *discordgo.Session, m *discordgo.MessageCreate) {
+	tempUsername, found := teamspeakTempUsernames[m.Author.ID]
+	ch, err := s.UserChannelCreate(m.Author.ID)
+	if err != nil {
+		s.ChannelMessageSend(m.ChannelID, "Something went wrong sending a DM to "+m.Author.Username)
+		return
+	}
+
+	// If a temp username has not been set yet we need to generate one and notify the user.
+	if found == false {
+		tmpUsername := randomString()
+		teamspeakTempUsernames[m.Author.ID] = tmpUsername
+		s.ChannelMessageSend(ch.ID, "Please adjust your username on teamspeak to: "+tmpUsername)
+		s.ChannelMessageSend(ch.ID, "When done please type !enable_mention again")
+		return
+	}
+
+	// If a temp username has been found we should check if the username is in use on teamspeak
+	tsUserId, found := getTeamspeakUserIdByName(tempUsername)
+	if found == false {
+		s.ChannelMessageSend(m.ChannelID, "Could not find a user with username: "+tempUsername)
+		return
+	}
+
+	// If a user has been found on teamspeak, grab its ID and add it to the discordTeamspeak mapping.
+	discordTeamspeakMapping[m.Author.ID] = strconv.Itoa(tsUserId)
+	delete(teamspeakTempUsernames, m.Author.ID)
+	jsonStr, err := json.Marshal(discordTeamspeakMapping)
+	if err != nil {
+		fmt.Printf("Error: %s", err.Error())
```

Comment:
You're swallowing an error and continuing. so if we can't save the user we still report success.